app/institutions/dashboard/-components/chart-kpi-wrapper/component.ts

```diff
@@ -185,91 +199,47 @@ export default class ChartKpiWrapperComponent extends Component<TotalCountChartW
      */
-    private calculateLicenses(licenseMetrics: InstitutionDepartmentModel[]): ChartDataModel[] {
-        const licenseData = [] as ChartDataModel[];
-
-        licenseMetrics.forEach((metric: InstitutionDepartmentModel ) => {
-            licenseData.push(
-                {
-                    label: metric.name,
-                    total: metric.numberOfUsers,
-                } as ChartDataModel,
-            );
+    @task
+    @waitFor
+    private async getShareData(
+        propertyPath: string,
+        index: number,
+        translationKey: string,
+        graphType?: string,
+    ) {
+        const valueSearch = await this.store.queryRecord('index-value-search', {
+            cardSearchFilter: {
+                affiliation: this.args.model.institution.iris.join(','),
+            },
+            sort: '-relevance',
+            'page[size]': 10,
+            valueSearchPropertyPath: propertyPath,
         });
-        return licenseData;
-    }
+        const resultPage = valueSearch.searchResultPage.toArray();
 
-    /**
-     * calculateAddons
-     *
-     * @description Abstracted method to build the ChartData model for add-ons
-     * @param addonMetrics The add-on metrics object
-     *
-     * @returns The add-ons ChartData model
-     */
-    private calculateAddons(addonMetrics: InstitutionDepartmentModel[]): ChartDataModel[] {
-        const addonData = [] as ChartDataModel[];
-
-        addonMetrics.forEach((metric: InstitutionDepartmentModel ) => {
-            addonData.push(
-                {
-                    label: metric.name,
-                    total: metric.numberOfUsers,
-                } as ChartDataModel,
-            );
-        });
-        return addonData;
-    }
+        const chartData = resultPage.map((result: SearchResultModel) => ({
+            total: result.cardSearchResultCount,
+            label: result.indexCard.get('label'),
+        }));
 
-    /**
-     * calculateStorageRegions
-     *
-     * @description Abstracted method to build the Storage Regions ChartData
-     * @param storageRegionsMetrics The storage regions metrics object
-     *
-     * @returns The storage regions ChartData model
-     */
-    private calculateStorageRegions(storageRegionsMetrics: InstitutionDepartmentModel[]): ChartDataModel[] {
-        const storageRegionsData = [] as ChartDataModel[];
+        const kpiChartObject = {
+            title: this.intl.t('institutions.dashboard.kpi-chart.' + translationKey),
+            chartData,
+            chartType: graphType ? graphType : 'bar',
 
-        storageRegionsMetrics.forEach((metric: InstitutionDepartmentModel ) => {
-            storageRegionsData.push(
-                {
-                    label: metric.name,
-                    total: metric.numberOfUsers,
-                } as ChartDataModel,
-            );
-        });
-        return storageRegionsData;
-    }
+        };
+        this.kpiCharts[index] = kpiChartObject;
```

Comment:
this is getting a bit convoluted and brittle, overwriting an object in an array by constant index (especially when `title` and `chartType` are unchanged but computed two ways) and then re-rendering everything that depends on that array (multiple times, it seems?)

i gather you don't want to make the whole set of charts block on these value-search requests, which makes sense... might simplify things if the `ChartKpiWrapper::ChartKpi` component understood the task instance you're passing it and automatically used `this.args.data.taskInstance.value` (when `this.args.data.chartData` is missing) -- then this task could have just one argument (`propertyPath`), return its `chartData`, and not have to do anything with `this.kpiCharts`

Reply:
Ahh, this is a much more sensible way to implement this. Will rework these components now